### ISSUE TITLE
🤖 Cuda Device Selection Support for Multi-GPU Machines 

### ISSUE THREAD

**Issue by Bug-Reaper** [pixray/pixray#53]
Threw a pretty [easy PR](https://github.com/pixray/pixray/pull/52) up which implements the feature described by post title:

I know it's not perfect as I think maybe other non-pixray libs may still default to the first cuda device. That said I've tested it and it seems to offload most of the heavy duty work to the cuda device selected via a new pixray argument.

Adds new pixray argument `-cud`, `--cuda_device`, `cuda_device`. 
`vq_parser.add_argument("-cud",  "--cuda_device", type=str, help="The Cuda device you want to use. (Typically 'cuda:0','cuda:1','cuda:2', etc...)", default='cuda:0', dest='cuda_device')`

Defaults to 'cuda:0' and thereby default behavior of pixray.py remains unchanged. It's essentially an extra feature for other people to use if they need it :100:

**Comment by Bug-Reaper** [comment]
Discovered yesterday that this new cuda device selection strategy breaks if you try to do edge loss. Basically edge loss seems to put some tensors on the first GPU device available no matter what and then complains about tensors split across multiple GPUs.

I think it should be somewhat straight forward to make edge loss respect the cuda_device arg though, I can add this improvement as soon as I have a chance and anyone else is welcome to add the improvement also if they can beat me to it :+1:

**Comment by dribnet** [comment]
Yeah - I believe the 'device' setting is passed to all modules at init time, but some of them might not be honouring it. I can check on this and patch it up in the next day or so.
